e2e/playwright/copy-move-actions/src/tests/copy.test.ts

```diff
@@ -0,0 +1,221 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ApiClientFactory, test, Utils, PersonalFilesPage, NodesApi, LoginPage } from '@alfresco/playwright-shared';
+import { expect } from '@playwright/test';
+import { Logger } from '@alfresco/adf-testing';
+
+test.describe('Copy actions', () => {
+  let nodesApi: NodesApi;
+  const apiClientFactory = new ApiClientFactory();
+  const username = `user-${Utils.random()}`;
+
+  let sourceFile: string;
+  let sourceFileInsideFolder: string;
+  let sourceFolder: string;
+  let destinationFolder: string;
+
+  let sourceFileId: string;
+  let sourceFileInsideFolderId: string;
+  let sourceFolderId: string;
+  let destinationFolderId: string;
+
+  test.beforeAll(async () => {
+    try {
+      await apiClientFactory.setUpAcaBackend('admin');
+      await apiClientFactory.createUser({ username });
+      nodesApi = await NodesApi.initialize(username, username);
+    } catch (error) {
+      Logger.error(`beforeAll failed : ${error}`);
+    }
+  });
+
+  test.afterAll(async ({ nodesApiAction }) => {
+    try {
+      await nodesApiAction.deleteCurrentUserNodes();
+    } catch (error) {
+      Logger.error(`afterAll failed : ${error}`);
+    }
+  });
+
+  test.beforeEach(async ({ personalFiles, page }) => {
+    sourceFile = `source-file-${Utils.random()}.txt`;
+    sourceFileInsideFolder = `source-file-inside-folder-${Utils.random()}.txt`;
+    sourceFolder = `source-folder-${Utils.random()}`;
+    destinationFolder = `destination-folder-${Utils.random()}`;
+
+    const loginPage = new LoginPage(page);
+    try {
+      await loginPage.loginUser(
+        { username, password: username },
+        {
+          withNavigation: true,
+          waitForLoading: true
+        }
+      );
+      destinationFolderId = (await nodesApi.createFolder(destinationFolder)).entry.id;
+      sourceFolderId = (await nodesApi.createFolder(sourceFolder)).entry.id;
+      sourceFileInsideFolderId = (await nodesApi.createFile(sourceFileInsideFolder, sourceFolderId)).entry.id;
+      sourceFileId = (await nodesApi.createFile(sourceFile)).entry.id;
+
+      await personalFiles.navigate();
+    } catch (error) {
+      Logger.error(`beforeEach failed : ${error}`);
+    }
+  });
+
+  const copyContentInPersonalFiles = async (personalFilesPage: PersonalFilesPage, sourceFileList: string[], destinationName: string) => {
+    personalFilesPage.copyOrMoveContentInDatatable(sourceFileList, destinationName, 'Copy');
```

Comment:
probably missing await as copyOrMoveContentInDatatable returns Promise

Reply:
Added missing await to copyOrMoveContentInDatatable